jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java

```diff
@@ -1131,11 +1131,15 @@ private void addVariableNameAndVariableValueParameters(TaskSummaryQueryBuilder q
         List<TaskEvent> taskEvents = commandService.execute(
     				new QueryNameCommand<List<TaskEvent>>("getAllTasksEvents", params));
 
-        if(taskEvents == null || taskEvents.isEmpty()) {
-            
-            UserTaskInstanceDesc task = getTaskById(taskId);
-            if (task == null) {
-                throw new TaskNotFoundException( MessageFormat.format(TASK_NOT_FOUND, taskId) );
+        
+        UserTaskInstanceDesc task = getTaskById(taskId);
+        if (task == null) {
+            throw new TaskNotFoundException( MessageFormat.format(TASK_NOT_FOUND, taskId) );
+        }
```

Comment:
This check can now be made in the beginning of the method as it would eliminate possibly unnecessary querying of all task events in case the task doesn't even exist.

Reply:
yes, good point